curricula/tests/test_user_permissions.py

```diff
@@ -0,0 +1,353 @@
+from django.test import TestCase
+
+from django.contrib.auth.models import Permission
+
+from lessons.factories import LessonFactory
+from curricula.factories import CurriculumFactory, UnitFactory, ChapterFactory, UserFactory, GroupFactory
+
+# Tests for admin pages in lessons/admin.py
+class UserPermissionsTestCase(TestCase):
+
+    def setUp(self):
+        # Create user groups with permissions
+        author_group = self.create_author_group()
+
+        partner_group = self.create_partner_group()
+
+        # Set up users
+        partner_user = UserFactory()
+        partner_user.is_staff = True
+        partner_group.user_set.add(partner_user)
+        partner_user.save()
+        self.partner_user = partner_user
+        self.assertTrue(self.client.login(username=partner_user.username, password='password'))
+
+        author_user = UserFactory()
+        author_user.is_staff = True
+        author_group.user_set.add(author_user)
+        author_user.save()
+        self.author_user = author_user
+        self.assertTrue(self.client.login(username=author_user.username, password='password'))
+
+        # Create curriculum, unit, chapter, lesson
+        self.partner_curriculum = CurriculumFactory(user=partner_user, slug="partner-curriculum")
+        self.partner_unit = UnitFactory(user=partner_user, parent=self.partner_curriculum, slug="partner-unit")
+        self.partner_chapter = ChapterFactory(user=partner_user, parent=self.partner_unit, slug="partner-chapter")
+        self.partner_lesson = LessonFactory(user=partner_user, parent=self.partner_unit)
+
+        self.author_curriculum = CurriculumFactory(user=author_user, slug="author-curriculum")
+        self.author_unit = UnitFactory(user=author_user, parent=self.author_curriculum, slug="author-unit")
+        self.author_chapter = ChapterFactory(user=author_user, parent=self.author_unit, slug="author-chapter")
+        self.author_lesson = LessonFactory(user=author_user, parent=self.author_unit)
+
+    # is_editable
+    def test_partner_curriculum_is_editable_by_partner(self):
+        request = self.client.get('/partner-curriculum/')
+        request.user = self.partner_user
+        self.assertTrue(self.partner_curriculum.is_editable(request))
+
+    def test_partner_curriculum_is_editable_by_author(self):
+        request = self.client.get('/partner-curriculum/')
```

Comment:
I don't think this line is doing anything, because is_editable seems to only be looking at request.user to make its decisions. try removing lines that look like this?

Reply:
I think is_editable needs something of that HttpResone type. To make it more clear I just switched it to get on the empty string. Does that seem alright? Or do you have another idea?